plugin_system/plugins/gregor_plugin.py

```diff
@@ -0,0 +1,144 @@
+import json
+import os
+import pandas as pd
+import pysam
+
+from plugin_system.plugins.base_plugin import BasePlugin
+
+
+class GregorPlugin(BasePlugin):
+    """
+    Plugin for GREGoR U08 release data on Terra
+    """
+
+    def create_sample_phenotype_index(
+        self, phenotype_table: str = None, as_set: bool = True, save_path: str = None
+    ) -> dict[str, list[str] | set[str]]:
+        """given phenotypical data input specified by the GREGoR Data model (in either tsv/csv or Terra data table),
+        return a dict mapping from each sample id to the sample's list of phenotypes
+
+        Returns:
+            dict[str, list[str]]: index of a sample id to sample's phenotypes
+        """
+
+        if phenotype_table is None:
```

Comment:
another consideration: it might be nice to try to refactor out code from here that does generically helpful tasks (e.g. grabbing a terra data table) and putting them into a `utils` subpackage so that other plugins wouldn't need to reimplement them

Reply:
Yes good idea, a util subpackage would be really helpful!

Reply:
very true good idea

Reply:
done